lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -137,57 +162,159 @@ static LogicalResult createPoolingOp(
                              ValueRange{paddedInput, windowTensor},
                              outTensorInitialized, stridesAttr, dilationAttr)
                .getResult(0);
-
   return success();
 }
 
-
 namespace {
-class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
+template <typename OpTy, int Dim>
+class ConvertAtenMaxPoolOp : public OpConversionPattern<OpTy> {
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+
+private:
+  LogicalResult writePoolingMax3D(AtenMaxPool3dOp &op,
+                                  typename OpTy::Adaptor &adaptor,
+                                  ConversionPatternRewriter &rewriter,
+                                  TypedAttr &smallestFPValueAttr, Value &self,
+                                  Type &elementType,
```

Comment:
Same comment for `elementType`. You can get this from `op`, which is already a parameter